### ISSUE TITLE
False Negative for no-static-element-interactions?

### ISSUE THREAD

**Issue by amilajack** [evcohen/eslint-plugin-jsx-a11y#89]
The following code throws an error:

```js
<a // <------ Visible, non-interactive elements should not have mouse or keyboard event listeners
  className="navbar-brand"
  onClick={this.props.setActiveMode.bind(this, 'movies')}
>
  Popcorn Time
</a>
```

Is that by design?

**Comment by ljharb** [comment]
Yes - a link should have an `href`. If you want an `onClick`, it should be a `<button type="button">`.

**Comment by amilajack** [comment]
I see. Thanks for clarifying.

**Comment by niksajanjic** [comment]
I had to turn off this rule for one specific case. Not that this rule should be changed, but it is too strict to be able to work in all cases.

If I have **tr** tag that needs to be clickable I will have this issue. Problem is that I can't wrap **tr** tag in **a** or **button** and only allowed child is **td** or **th**. This is by specification. So, there's only 1 hackish way of making **tr** clickable:

```
<table>
  <tbody>
    <tr>
      <td><a href="/user/1">John Smith</a></td>
      <td><a href="/user/1">123 Fake St</a></td>
      <td><a href="/user/1">90210</a></td>
    </tr>
    <tr>
      <td><a href="/user/2">Peter Nguyen</a></td>
      <td><a href="/user/2">456 Elm Ave</a></td>
      <td><a href="/user/2">90210</a></td>
    </tr>
  </tbody>
</table>
```

This is of course bad way of doing it and it's not acceptable for us. Also, there's a way of not using a table, but instead using div's and styling them to look like table using CSS, but again, that is not option as it breaks semantics.

**Comment by Tuxiaochao** [comment]
ts

**Comment by RShergold** [comment]
Since this is the top hit on google when searching for the error message. Adding `tabIndex` will make `a` a valid interactive element.

eg:
```
<a onClick={ func } tabIndex={ 0 }>
````
